### PR TITLE
Add depth/stencil GPUTextureFormats

### PR DESCRIPTION
**Pull request by JusSn**
Follow-up to #206. I didn't see @Kangz suggestion that we just infer these until I'd already created the PR so 🤷‍♂️

### CODE REVIEW

File: design/sketch.webidl [hunk 1]
```diff
@@ -147,6 +147,12 @@ enum GPUTextureFormat {
     "rgba32uint",
     "rgba32sint",
     "rgba32float",
+    /* Depth and Stencil formats */
+    "depth16unorm",
+    "depth32float",
+    "stencil8",
```
Comment (RafaelCintron) [inline design/sketch.webidl]: Unless I am missing something, D3D12 does not natively support 'stencil8'. 

The core standard should only contain formats supported by all APIs.

Comment (JusSn) [reply design/sketch.webidl]: Updated, thanks! The rest look good on the Vk side, from my understanding.

**Comment by kvark** [comment]
Vulkan story is fun: *one* of a list of formats has to support being a depth/stencil attachment:
>VK_FORMAT_FEATURE_DEPTH_STENCIL_ATTACHMENT_BIT feature must be supported for at least one of VK_FORMAT_X8_D24_UNORM_PACK32 and VK_FORMAT_D32_SFLOAT, and must be supported for at least one of VK_FORMAT_D24_UNORM_S8_UINT and VK_FORMAT_D32_SFLOAT_S8_UINT.

This is quite annoying. It would be good to find out what the actual support is for the GPUs.

**Comment by magcius** [comment]
AFAIUI, combined depth/stencil formats are basically an API-level lie, and no hardware actually combines them, hence what might seem like a bizarre 32/8 format. I imagine almost all hardware will be able to support that.

http://vulkan.gpuinfo.org/listformats.php suggests that 32/8 is supported by 99% of reported Vulkan implementations, and 24/8 is supported by 75% of them.

**Comment by kvark** [comment]
@magcius I see 32/8 (with optimal tiling) *not* being supported on a [solid portion of devices](http://vulkan.gpuinfo.org/listreports.php?optimalformat=D32_SFLOAT_S8_UINT&option=not). Given that `S8` isn't available on D3D12, we end up with a situation where there is no format with stencil that is supported everywhere...

From the user perspective, in this day and age, I believe D32F_S8 is critical to have: a typical high-performance 3D application would want to use D32F for the reverse-Z depth and S8 for masking out things. Thinking of what we can do to support it on platforms that don't have it natively... these are all Android devices, so we only need Vulkan backend to support this path. This is a good thing, since on Vulkan we can have different surfaces for depth and stencil attachments, so we could manage stencil as a separate texture, internally, as long as the user never needs to sample from both of them (as different channels of a texture view).

**Comment by JusSn** [comment]
Metal story is more complicated than I thought. Depth16unorm and Depth24_stencil8 aren't available on iOS, only Mac. I'll update.

**Comment by kainino0x** [comment]
Huh, there are only 32-bit depth formats on iOS? (I think you're right based on the [tables](https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf).) That seems like a waste of bandwidth - AFAIK lots of apps don't need that much precision.

**Pull request merged**